src/cata_tiles.h

```diff
@@ -126,6 +128,78 @@ struct tile_drawing_cache {
     }
 };
 
+struct pixel {
+    int r;
+    int g;
+    int b;
+    int a;
+
+    pixel()
+    {
+        r = 0;
+        g = 0;
+        b = 0;
+        a = 0;
+    }
+
+    pixel(SDL_Color c)
+    {
+        r = c.r;
+        g = c.g;
+        b = c.b;
+        a = c.a;
+    }
+
+    SDL_Color getSdlColor()
+    {
+        SDL_Color c;
+        c.r = static_cast<Uint8>(r);
+        c.g = static_cast<Uint8>(g);
+        c.b = static_cast<Uint8>(b);
+        c.a = static_cast<Uint8>(a);
+        return c;
+    }
+
+    bool isBlack()
+    {
+        return (r == 0 && g == 0 && b == 0);
+    }
+
+    operator==(const pixel &other)
+    {
+        return (r == other.r && g == other.g && b == other.b && a == other.a);
+    }
+    operator!=(const pixel &other)
```

Comment:
This kind of comparison operators should be const. Similar the `isBlack` function should be const as well.